app/models/manageiq/providers/redfish/physical_infra_manager/operations/power.rb

```diff
@@ -42,24 +42,43 @@ def restart_mgmt_controller(_server, _options)
 
     private
 
-    def reset_server(server, reset_type)
-      $redfish_log.info("Requesting #{reset_type} for #{server.ems_ref}.")
-      with_provider_connection do |client|
-        system = client.find(server.ems_ref)
-        if system.nil?
-          $redfish_log.error("#{server.ems_ref} does not exist anymore.")
-          return
+    def execute_first_valid_power_action(server, rtypes)
```

Comment:
Just realized this is potentially async operation, which is fine=yes, but perhaps we should be more explicit about it, like

```
execute_first_valid_power_action_async
```

or
```
trigger_first_valid_power_action
```

Reply:
Will user the trigger variant.